### PR TITLE
Misc

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 package io.fabric8.kubernetes.client.dsl.base;
 
 import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.zjsonpatch.internal.guava.Strings;
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java]: Really? Shouldn't really use internal packages, especially for OSGi.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java [hunk 2]
```diff
@@ -325,6 +326,9 @@ public T apply(T resource) {
 
   @Override
   public T createOrReplace(T item) {
+    if (Strings.isNullOrEmpty(name) && item instanceof HasMetadata) {
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java]: This seems unrelated?

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicaSetOperationsImpl.java [hunk 3]
```diff
@@ -133,13 +133,18 @@ private void waitUntilScaled() {
...
       public void run() {
-        ReplicaSet rc = getMandatory();
-        atomicRC.set(rc);
-        if (rc.getStatus().getObservedGeneration() >= rc.getMetadata().getGeneration() && Objects.equals(rc.getSpec().getReplicas(), rc.getStatus().getReplicas())) {
-          countDownLatch.countDown();
-        } else {
+        try {
+          ReplicaSet rc = getMandatory();
+          atomicRC.set(rc);
+          if (rc.getStatus() == null || rc.getStatus().getObservedGeneration() == null || rc.getMetadata().getGeneration() == null) {
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicaSetOperationsImpl.java]: Bad formatting, plus shouldn't have an empty `if` block. Just add it to the conditional below.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicationControllerOperationsImpl.java [hunk 4]
```diff
@@ -151,13 +151,17 @@ private void waitUntilRCIsScaled() {
...
-        ReplicationController rc = getMandatory();
-        atomicRC.set(rc);
-        if (Objects.equals(rc.getSpec().getReplicas(), rc.getStatus().getReplicas())) {
-          countDownLatch.countDown();
-        } else {
-          LOG.debug("Only {}/{} replicas scheduled for ReplicationController: {} in namespace: {} seconds so waiting...",
+        try {
+          ReplicationController rc = getMandatory();
+          atomicRC.set(rc);
+          if (Objects.equals(rc.getSpec().getReplicas(), rc.getStatus().getReplicas())) {
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicationControllerOperationsImpl.java]: This will countdown if both are null - is that what you want to do?

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java (hunk 2 above)
Comment (iocanel) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java]: I needed that in the example.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicationControllerOperationsImpl.java (hunk 4 above)
Comment (iocanel) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicationControllerOperationsImpl.java]: The line in question is not mine. I just wrapped the code in try catch and reformatted.

Is it valid to have null as replicas?

Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicationControllerOperationsImpl.java]: There's nothing defending against that here but it shouldn't be possible via remote API I don't think.

**Comment by jimmidyson** [comment]
[merge]
